operator/src/main/java/org/bf2/operator/operands/AbstractKafkaCluster.java

```diff
@@ -426,39 +429,58 @@ protected List<GenericKafkaListener> buildListeners(ManagedKafka managedKafka, i
-                                .withPort(9096)
-                                .withType(KafkaListenerType.INTERNAL)
-                                .withNetworkPolicyPeers(new NetworkPolicyPeerBuilder()
-                                        .withNewPodSelector()
-                                            .addToMatchLabels("strimzi.io/name", managedKafka.getMetadata().getName() + "-kafka")
-                                        .endPodSelector()
-                                        .build())
-                                .withTls(false)
-                                .build()
-                );
+        List<GenericKafkaListener> listeners = Arrays.asList(
+                new GenericKafkaListenerBuilder()
+                        .withName(EXTERNAL_LISTENER_NAME)
+                        .withPort(9094)
+                        .withType(externalListenerType)
+                        .withTls(true)
+                        .withAuth(plainOverOauthAuthenticationListener)
+                        .withConfiguration(listenerConfigBuilder.build())
+                        .build(),
+                new GenericKafkaListenerBuilder()
+                        .withName("oauth")
+                        .withPort(9095)
+                        .withType(KafkaListenerType.INTERNAL)
+                        .withTls(true)
+                        .withAuth(oauthAuthenticationListener)
+                        .withNetworkPolicyPeers(new NetworkPolicyPeerBuilder()
+                                .withNewPodSelector()
+                                .addToMatchLabels("app", AbstractAdminServer.adminServerName(managedKafka))
+                                .endPodSelector()
+                                .build())
+                        .build(),
+                new GenericKafkaListenerBuilder()
+                        .withName("sre")
+                        .withPort(9096)
+                        .withType(KafkaListenerType.INTERNAL)
+                        .withNetworkPolicyPeers(new NetworkPolicyPeerBuilder()
+                                .withNewPodSelector()
+                                .addToMatchLabels("strimzi.io/name", managedKafka.getMetadata().getName() + "-kafka")
+                                .endPodSelector()
+                                .build())
+                        .withTls(false)
+                        .build()
+        );
+
+        // Apply any override configuration to listeners
+        var strimzi = managedKafka.getSpec().getVersions().getStrimzi();
+        if (overrideManager.getKafkaOverride(strimzi) != null) {
+            var objectMapper = new ObjectMapper();
+            listeners.forEach(listener -> {
+                var listenerOverride = overrideManager.getKafkaOverride(strimzi).getListeners().getOrDefault(listener.getName(), new ListenerOverride());
+                if (listener.getAuth() != null && listenerOverride.getAuth() != null) {
```

Comment:
Cosmetic, but it might be a bit clearer to filter out those listeners without auth overrides using a stream.
e.g.

```java
listeners.stream()
         .filter(listener -> listener.getAuth() != null)
         .filter(listener -> listenerHasAuthOverrides(listener))
         .forEach(listener -> { ... });
```

Reply:
Thanks @MikeEdgar , I've implemented your suggestion.